src/VecSim/vec_sim_index.h

```diff
@@ -48,6 +48,15 @@ class VecSimIndex : public VecsimBaseObject {
      */
     virtual size_t indexSize() const = 0;
 
+    /**
+     * @brief Resolves VecSimRawParam array and generate VecSimQueryParams struct.
+     * @param index the index whose size is requested.
+     * @param rparams array of raw params to resolve.
```

Comment:
you choose to take the assumption that you will get an instance of `arr.h` this should not be the basic assumption. In C API, you give a pointer and a length of the array